torch/distributed/_tensor/_dispatch.py

```diff
@@ -231,6 +236,13 @@ def default_tensor(spec: DTensorSpec) -> torch.Tensor:
             if output_sharding.output_spec is not None:
                 return args[0]
             else:
+                if op_call == aten._amp_foreach_non_finite_check_and_unscale_.default:
+                    found_inf_dtensor = dtensor.DTensor.from_local(
+                        args[1], mesh, [Partial("max")]
```

Comment:
if we decide on DTensor found_inf, maybe add a 2D unit test? the hardcoded 1-dim Partial(“max”) looks risky. the placement should be same to args00 (grad)

Reply:
For 2D, we will have to assume placements[0] is the `Partial(“max”)`, and `placements[1] is Replicate()`. So we can make this generic such that we always make placements[0] partial. This should work for both our 1D and 2D cases for now.

Reply:
Later we found that both placements should be `Partial(“max”)`.
For example, if we found_inf at rank 0, then [`Partial(“max”)`, `Replicate()`] would only make rank1 to have 1 and rank2,3 to have 0
Similarly, [`Replicate()`, `Partial(“max”)`] would only make rank2 to have 1 and rank1,3 to have 0

Reply:
> Later we found that both placements should be `Partial(“max”)`. For example, if we found_inf at rank 0, then [`Partial(“max”)`, `Replicate()`] would only make rank1 to have 1 and rank2,3 to have 0 Similarly, [`Replicate()`, `Partial(“max”)`] would only make rank2 to have 1 and rank1,3 to have 0

Just realized the issue is that when we are using `from_local`, there is no additional check for local_tensor. Users would need to make sure that the local tensor is replicated on the Replicate() dimension. In a 2x2 case, rank 0 and 1 would need to have the same local tensor, so as the rank 2 and 3.

Reply:
found_inf is decided by .grad. if gradients are truly replicated (enforced by parallelsims), found_inf are replicated. That’s why I keep mentioning  constructing found_inf from args00.new_full

Reply:
Now apply the placements following args00.placements